state/accountsDB.go

```diff
@@ -542,20 +547,20 @@ func (adb *AccountsDB) saveAccountToTrie(accountHandler vmcommon.AccountHandler)
 		return err
 	}
 
-	return adb.mainTrie.Update(accountHandler.AddressBytes(), buff)
+	return mainTrie.Update(accountHandler.AddressBytes(), buff)
 }
 
 // RemoveAccount removes the account data from underlying trie.
 // It basically calls Update with empty slice
 func (adb *AccountsDB) RemoveAccount(address []byte) error {
-	adb.mutOp.Lock()
-	defer adb.mutOp.Unlock()
-
 	if len(address) == 0 {
 		return fmt.Errorf("%w in RemoveAccount", ErrNilAddress)
 	}
 
-	acnt, err := adb.getAccount(address)
+	adb.mutOp.Lock()
```

Comment:
Please add a comment stating this is a critical section

Reply:
Done.